src/m365/entra/commands/group/group-set.spec.ts

```diff
@@ -0,0 +1,367 @@
+    const actual = await command.validate({ options: { displayName: 'Cli group', ownerUserNames: userUpns.join(','), memberUserNames: userUpns.join(',') } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation if no options to be updated is specified', async () => {
+    const actual = await command.validate({ options: { id: groupId } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('successfully updates group specified by id', async () => {
+    sinon.stub(request, 'get').resolves({ value: [] });
+    const patchRequestStub = sinon.stub(request, 'patch').resolves();
+
+    await command.action(logger, { options: { id: groupId, description: 'Microsoft 365 group', mailNickname: 'Microsoft365Group', visibility: 'Public', newDisplayName: '365 group', verbose: true } });
+    assert.deepStrictEqual(patchRequestStub.lastCall.args[0].data, updateGroupRequest);
+  });
+
+  it('successfully updates group specified by displayName', async () => {
+    sinon.stub(request, 'get').resolves({ value: []});
+    sinon.stub(entraGroup, 'getGroupIdByDisplayName').withArgs('Microsoft 365 Group').resolves(groupId);
+
+    const patchRequestStub = sinon.stub(request, 'patch').resolves();
+
+    await command.action(logger, { options: { displayName: 'Microsoft 365 Group', description: 'Microsoft 365 group', mailNickname: 'Microsoft365Group', visibility: 'Public', newDisplayName: '365 group' } });
+    assert.deepStrictEqual(patchRequestStub.lastCall.args[0].data, updateGroupRequest);
+  });
+
+  it('successfully updates group with owners specified by ids and removes current owners', async () => {
+    sinon.stub(request, 'get').resolves({ value: [{ id: '717f1683-00fa-488c-b68d-5d0051f6bcfa' }] });
+    sinon.stub(request, 'patch').resolves();
+
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch' &&
+        opts.data.requests[0].method === 'PATCH') {
+        return {
+          responses: Array(2).fill({
+            status: 204,
+            body: {}
+          })
+        };
+      }
+
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch' &&
+        opts.data.requests[0].method === 'DELETE') {
+        return {
+          responses: [
+            {
+              status: 204,
+              body: {}
+            }
+          ]
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { id: groupId, description: 'Microsoft 365 group', mailNickname: 'Microsoft365Group', visibility: 'Public', ownerIds: userIds.join(',') } });
+    assert.deepStrictEqual(postStub.firstCall.args[0].data.requests, addOwnersRequest);
+  });
+
+  it('successfully updates group with members specified by user names and removes current members', async () => {
+    sinon.stub(request, 'get').resolves({ value: [{ id: '717f1683-00fa-488c-b68d-5d0051f6bcfa' }] });
```

Comment:
Why is this stub needed?

Reply:
It was added due to reading users of a group

    const groupUsers = await odata.getAllItems<User>...

Reply:
In that case, let's validate the URL as well.